src/map/lua/lua_baseentity.cpp

```diff
@@ -16660,6 +16679,40 @@ bool CLuaBaseEntity::hasTPMoves()
     return !MobSkills.empty();
 }
 
+/************************************************************************
+ *  Function: drawIn()
+ *  Purpose : Draws in the target, or current target if not specified
+ *  Example : mob:drawIn()     mob:drawIn(player)
+ *  Notes   : Draws in a player even if within the draw-in leash
+ ************************************************************************/
+void CLuaBaseEntity::drawIn(sol::variadic_args va)
+{
+    auto mobObj = dynamic_cast<CMobEntity*>(m_PBaseEntity);
+
+    if (va.size() == 0)
+    {
+        auto defaultTarget = mobObj->GetBattleTarget();
+
+        if (defaultTarget == nullptr)
+        {
+            return;
+        }
+        battleutils::DrawIn(defaultTarget, mobObj, mobObj->GetMeleeRange() - 0.2f);
+        return;
+    }
+
+    CBattleEntity*  PTarget{ nullptr };
+    CLuaBaseEntity* PLuaBaseEntity = va.get<CLuaBaseEntity*>(0);
+    PTarget                        = PLuaBaseEntity ? (CBattleEntity*)PLuaBaseEntity->m_PBaseEntity : nullptr;
+
+    if (PTarget)
+    {
+        battleutils::DrawIn(PTarget, mobObj, mobObj->GetMeleeRange() - 0.2f);
```

Comment:
Where does this 0.2f come from? What happens if GetMeleeRange returns zero? (I haven't looked to see if that's a possibility)

Reply:
The `GetMeleeRange() - 0.2f` is from looking at how Draw-In was done from the `mob_controller.cpp` file
Line 694 - 701 in `mob_controller.cpp`:
```cpp
            if (PMob->getMobMod(MOBMOD_DRAW_IN) > 0)
            {
                if (currentDistance >= PMob->GetMeleeRange() * 2 && battleutils::DrawIn(PTarget, PMob, PMob->GetMeleeRange() - 0.2f))
                {
                    FaceTarget();
                    return;
                }
            }
```